tests/it/main.rs

```diff
@@ -252,6 +252,24 @@ fn test_pushenv_lock() {
     t2.join().unwrap();
 }
 
+#[test]
+#[should_panic]
+fn test_cannot_use_output_with_ignored_stdout() {
+    let _ = cmd!("asdf").ignore_stdout().read();
+}
+
+#[test]
+#[should_panic]
+fn test_cannot_use_output_with_ignored_stderr() {
+    let _ = cmd!("asdf").ignore_stderr().read_stderr();
+}
+
+#[test]
+#[should_panic]
+fn test_cannot_use_output_with_ignored_both() {
+    let _ = cmd!("adsf").ignore_stdout().ignore_stderr().read_stderr();
+}
+
```

Comment:
Let's add a positive, non-panicing test here: that ignoring stdout actually ignores stdout, and that, while you ignore stdout, you actually can read out stderr.

Reply:
How should I write to stderr without using shell specific features? Is there also a way to detect that ignoring a stream does do that?